### DISCUSSION TITLE
pCellEntityCall UE4就崩溃了

### DISCUSSION THREAD

**Discussion by a492486** [kbengine/kbengine#1512] [General]
@kebiao 大佬帮忙瞅瞅

avatar下函数就调用了下pCellEntityCall UE4就崩溃了，查看报错位置	ScriptModule** smFind = EntityDef::moduledefs.Find(className);

Unhandled Exception: EXCEPTION_ACCESS_VIOLATION reading address 0x00000018
UE4Editor_KBEnginePlugins!TSet<TTuple<FString,KBEngine::ScriptModule *>,TDefaultMapHashableKeyFuncs<FString,KBEngine::ScriptModule *,0>,FDefaultSetAllocator>::FindId() [F:\ue4\UE_4.25\Engine\Source\Runtime\Core\Public\Containers\Set.h:776]
UE4Editor_KBEnginePlugins!KBEngine::EntityCall::newCall() [F:\kbengine\kbengine_ue4_demo-2.5.8\Plugins\kbengine_ue4_plugins\Source\KBEnginePlugins\Engine\EntityCall.cpp:48]
UE4Editor_KBEnginePlugins!KBEngine::EntityCellEntityCall_AvatarBase::hasSkillList() [F:\kbengine\kbengine_ue4_demo-2.5.8\Plugins\kbengine_ue4_plugins\Source\KBEnginePlugins\Engine\EntityCallAvatarBase.cpp:64]
UE4Editor_KBEnginePlugins!KBEngine::Avatar::__init__() [F:\kbengine\kbengine_ue4_demo-2.5.8\Plugins\kbengine_ue4_plugins\Source\KBEnginePlugins\Scripts\Avatar.cpp:35]
服务端
	def hasSkillList(self,exposed):
		"""
		exposed
		"""
		if exposed != self.id:
			return
		DEBUG_MSG("Avatar::hasSkillList: %i, skills_list=%s." % (self.id, len(self.skills)))

		return  self.client.onhasSkillList(self.skills)
Avatar.def
		<hasSkillList>
		    <Exposed/>
		</hasSkillList>

**Comment by basesrv** [comment]
客户端没实现这个模块吧



    
    
    



    
        
            
                
                    
                        
                    
                    
                        
                            kbe k
                            
                            
                                
                                    ***@***.***
                                
                        
                    
                
            
        
    


---- 回复的原邮件 ***@***.***>发送日期2021年10月04日 23:17 ***@***.***> ***@***.***>主题[kbengine/kbengine] pCellEntityCall UE4就崩溃了 (Discussion #1512)

@kebiao 大佬帮忙瞅瞅

avatar下函数就调用了下pCellEntityCall UE4就崩溃了，查看报错位置	ScriptModule** smFind = EntityDef::moduledefs.Find(className);

Unhandled Exception: EXCEPTION_ACCESS_VIOLATION reading address 0x00000018

UE4Editor_KBEnginePlugins!TSet<TTuple<FString,KBEngine::ScriptModule *>,TDefaultMapHashableKeyFuncs<FString,KBEngine::ScriptModule *,0>,FDefaultSetAllocator>::FindId() [F:\ue4\UE_4.25\Engine\Source\Runtime\Core\Public\Containers\Set.h:776]

UE4Editor_KBEnginePlugins!KBEngine::EntityCall::newCall() [F:\kbengine\kbengine_ue4_demo-2.5.8\Plugins\kbengine_ue4_plugins\Source\KBEnginePlugins\Engine\EntityCall.cpp:48]

UE4Editor_KBEnginePlugins!KBEngine::EntityCellEntityCall_AvatarBase::hasSkillList() [F:\kbengine\kbengine_ue4_demo-2.5.8\Plugins\kbengine_ue4_plugins\Source\KBEnginePlugins\Engine\EntityCallAvatarBase.cpp:64]

UE4Editor_KBEnginePlugins!KBEngine::Avatar::init() [F:\kbengine\kbengine_ue4_demo-2.5.8\Plugins\kbengine_ue4_plugins\Source\KBEnginePlugins\Scripts\Avatar.cpp:35]

服务端

def hasSkillList(self,exposed):

"""

exposed

"""

if exposed != self.id:

return

DEBUG_MSG("Avatar::hasSkillList: %i, skills_list=%s." % (self.id, len(self.skills)))

	return  self.client.onhasSkillList(self.skills)



Avatar.def









—You are receiving this because you are subscribed to this thread.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.



[

{

***@***.***": "http://schema.org",

***@***.***": "EmailMessage",

"potentialAction": {

***@***.***": "ViewAction",

"target": "https://github.com/kbengine/kbengine/discussions/1512",

"url": "https://github.com/kbengine/kbengine/discussions/1512",

"name": "View Discussion"

},

"description": "View this Discussion on GitHub",

"publisher": {

***@***.***": "Organization",

"name": "GitHub",

"url": "https://github.com"

}

}

]

**Reply by a492486** [reply]
客户端EntityDef初始化已经生成了，关键是写了两个，有一个正常，另一个一调UE4就没了。。
TArray<DATATYPE_BASE*> Avatar_hasSkillList_args;

	Method* pAvatar_hasSkillList = new Method();
	pAvatar_hasSkillList->name = TEXT("hasSkillList");
	pAvatar_hasSkillList->methodUtype = 6;
	pAvatar_hasSkillList->aliasID = -1;
	pAvatar_hasSkillList->args = Avatar_hasSkillList_args;

	pAvatarModule->methods.Add(TEXT("hasSkillList"), pAvatar_hasSkillList); 
	pAvatarModule->cell_methods.Add(TEXT("hasSkillList"), pAvatar_hasSkillList);

	pAvatarModule->idcell_methods.Add(pAvatar_hasSkillList->methodUtype, pAvatar_hasSkillList);

	//DEBUG_MSG("EntityDef::initScriptModules: add(Avatar), method(hasSkillList / 6).");

**Reply by a492486** [reply]
唯一的区别就是有问题的这个没传参

**Comment by a492486** [comment]
我把这个注册成事件就可以了，直接在Avatar的初始化方法里调就不行，打日志估计是因为初始化的时候还没有pCellEntityCall的classname还没写进去avatar，但是我看accout下的初始化方法是可以调的。。不知道问题出在哪里
